hathor/transaction/block.py

```diff
@@ -267,14 +286,16 @@ def verify_checkpoint(self, checkpoints: List[Checkpoint]) -> None:
         assert self.storage is not None
         meta = self.get_metadata()
         # XXX: it's fine to use `in` with NamedTuples
-        if Checkpoint(meta.height, self.hash) in checkpoints:
+        if Checkpoint(meta.get_soft_height(), self.hash) in checkpoints:
             return
         # otherwise at least one child must be checkpoint validated
-        for child_tx in map(self.storage.get_transaction, meta.children):
-            if child_tx.get_metadata().validation.is_checkpoint():
-                return
-        raise CheckpointError(f'Invalid new block {self.hash_hex}: expected to reach a checkpoint but none of '
-                              'its children is checkpoint-valid and its hash does not match any checkpoint')
+        # XXX: because we decided to remove "partial" children metadata, this won't work, we could add a new metadata
```

Comment:
Why should we keep this piece of commented code?

Reply:
I'll remove it and add the uncommented code on the next branch.

Reply:
Done! Thanks!